File: js/linearSearch.js
const arrayToSearchThrough = [];
for (let i = 1; i <= 1000; i++) {
  arrayToSearchThrough.push(i);
}

exports.linearSearch = function (valueToFind, arrayToSearchThrough) {
  let index;
  for (let x = 0; x < arrayToSearchThrough.length; x++) {
    if (arrayToSearchThrough[x] === valueToFind) {
      index = x;
    }
  }
  return index;
};

exports.linearSearchGlobal = function (valueToFind, arrayToSearchThrough) {
  let indices = [];
  for (let x = 0; x < arrayToSearchThrough.length; x++) {
    if (arrayToSearchThrough[x] === valueToFind) {
      indices.push(x);
    }
  }
  return indices;
};

File: python/linear_search.py
array_to_search_through = []
for number in range(1, 1001):
    array_to_search_through.append(number)

def linear_search(value_to_find, array_to_search_through):
    index = None

    for x in range(len(array_to_search_through)):
        if array_to_search_through[x] == value_to_find:
            index = x

    return index

def global_linear_search (value_to_find, array_to_search_through):
    indices = []

    for x in range(len(array_to_search_through)):
        if array_to_search_through[x] == value_to_find:
            indices.append(x)
            
    return indices
